src/AzureRMTools.ts

```diff
@@ -321,6 +286,145 @@ export class AzureRMTools {
         }
     }
 
+    private logDeploymentTemplateErrors(document: vscode.TextDocument, deploymentTemplate: DeploymentTemplate): void {
+        deploymentTemplate.errors
+            .then((parseErrors: language.Issue[]) => {
+                const diagnostics: vscode.Diagnostic[] = [];
+
+                for (const error of parseErrors) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, error, vscode.DiagnosticSeverity.Error));
+                }
+
+                for (const warning of deploymentTemplate.warnings) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, warning, vscode.DiagnosticSeverity.Warning));
+                }
+
+                this._diagnosticsCollection.set(document.uri, diagnostics);
+            })
+            .catch((error: any) => {
+                this.logError("UnhandledError", error);
+            });
+    }
+
+    /**
+     * Update survey settings on disk and return the updated version
+     */
+    private updateSurveySettingsFromOpeningTemplate(): { surveySettings: SurveySettings, surveySettingsFilePath: string } {
```

Comment:
I find this object being used at the call site, and then passed into `initializeExtension` 
Would making it an interface help?

Reply:
Implemented, although not sure it actually improved readability because a bunch of places now dereference twice.